### ISSUE TITLE
vendor_name missing from user_factor model. 

### ISSUE THREAD

**Issue by kevinklitzke** [okta/okta-sdk-python#330]
The vendorName attribute is missing from the user_factor model.

While this is often the same as the provider value for the built-in factors, however for custom TOTP/HOTP factors this attribute extremely useful as the provider value is simply "CUSTOM" and the vendorName is the descriptive value  provided when creating the custom factor. 

While you could drop down to the raw API to work around this, that is not an ideal solution.

**Comment by gabrielsroka** [comment]
here's a quick workaround. see also https://github.com/okta/okta-sdk-python/#call-other-api-endpoints

```python
import okta.client
import asyncio

async def main():
    async with okta.client.Client() as client:
        user_id = 'me'
        factor_id = 'chfwb...'

        # SDK doesn't support vendorName.
        factor, resp, _ = await client.get_factor(user_id, factor_id)
        print(factor)

        # Call API to get vendorName
        exec = client.get_request_executor()
        req, _ = await exec.create_request('get', f'/api/v1/users/{user_id}/factors/{factor_id}')
        resp, _ = await exec.execute(req)
        factor = resp.get_body()
        print(factor, factor['vendorName'])

asyncio.run(main())
```

**Comment by gabrielsroka** [comment]
here's a custom model:

```python
import okta.client
import okta.constants
import okta.models.factor_type
import okta.models.custom_hotp_user_factor
import asyncio

async def main():
    okta.constants.OKTA_FACTOR_TYPE_TO_FACTOR[okta.models.factor_type.FactorType.TOKEN_HOTP] = CustomHotpUserFactor
    async with okta.client.Client() as client:
        user_id = 'me'
        factor_id = 'chfwb...'
        factor, *_ = await client.get_factor(user_id, factor_id)
        print(factor, factor.vendorName)

class CustomHotpUserFactor(okta.models.custom_hotp_user_factor.CustomHotpUserFactor):
    def __init__(self, config=None):
        super().__init__(config)
        self.vendorName = config['vendorName']

asyncio.run(main())
```

**Comment by gabrielsroka** [comment]
here's another solution that probably fixes some problems, yet introduces new ones:

```python
import okta.client
import okta.constants
import okta.models.user_factor
import asyncio

async def main():
    for k in okta.constants.OKTA_FACTOR_TYPE_TO_FACTOR:
        okta.constants.OKTA_FACTOR_TYPE_TO_FACTOR[k] = UserFactor
    async with okta.client.Client() as client:
        user_id = 'me'

        factor_id = 'chfwbvu2goF5YdNR90h7'
        factor, *_ = await client.get_factor(user_id, factor_id)
        print(factor, factor.vendorName)

        factors, *_ = await client.list_factors(user_id)
        for factor in factors:
            print(factor, factor.vendorName)

class UserFactor(okta.models.user_factor.UserFactor):
    def __init__(self, config=None):
        super().__init__(config)
        self.vendorName = config['vendorName']

asyncio.run(main())
```

**Comment by bretterer** [comment]
This should be resolved in 2.9.0.  If it is still an issue for you, please open a new GH Issue referencing this one
